src/runtime_src/core/pcie/linux/scan.cpp

```diff
@@ -605,9 +606,43 @@ std::shared_ptr<pcidev::pci_device> pcidev::pci_device::lookup_peer_dev()
     return udev;
 }
 
+std::shared_ptr<pcidev::pci_device> pcidev::pci_device::lookup_parent_dev()
+{
+    std::string err;
+    std::shared_ptr<pcidev::pci_device> pdev;
+    boost::filesystem::path sys_path; 
+    uint16_t vendor;
+
+    if (!is_mgmt)
+        return NULL;
+
+    sys_path = get_sysfs_path("", "dparent");
+    if (!boost::filesystem::exists(sys_path) && !boost::filesystem::is_symlink(sys_path))
+        return NULL;
+
+    sys_path = boost::filesystem::read_symlink(sys_path);
+    pdev = std::make_shared<pcidev::pci_device>(
+            std::string(sys_path.filename().c_str()));
+
+    // Determine if device is a xilinx bridge
+    pdev->sysfs_get<uint16_t>("", "vendor", err, vendor,
+            static_cast<uint16_t>(-1));
+    if (!err.empty()) {
+        std::cout << err << std::endl;
+        return NULL;
+    }
+
+    if (vendor != XILINX_ID)
+        return NULL;
```

Comment:
Why you don't return parent when bridge is not from xilinx? How does it work for non-U.2 boards?